test/graphql/Offer/mutations/republishOffer.test.ts

```diff
@@ -0,0 +1,165 @@
+  };
+
+  it("republish an offer for students that expired", async () => {
+    await expectToRepublishTheOffer(offers[ApplicantType.student]);
+  });
+
+  it("republish an offer for graduates that expired", async () => {
+    await expectToRepublishTheOffer(offers[ApplicantType.graduate]);
+  });
+
+  it("republish an offer for both that expired", async () => {
+    await expectToRepublishTheOffer(offers[ApplicantType.both]);
+  });
+
+  it("returns an error if no user is logged in", async () => {
+    const { errors } = await performMutation(client.loggedOut(), {
+      uuid: offers[ApplicantType.student].uuid
+    });
+    expect(errors).toEqualGraphQLErrorType(AuthenticationError.name);
+  });
+
+  it("returns an error if the current user is an applicant", async () => {
+    const { apolloClient } = await TestClientGenerator.applicant();
+    const { errors } = await performMutation(apolloClient, {
+      uuid: offers[ApplicantType.both].uuid
+    });
+    expect(errors).toEqualGraphQLErrorType(UnauthorizedError.name);
+  });
+
+  it("returns an error if the current user is an approved applicant", async () => {
+    const { apolloClient } = await TestClientGenerator.applicant({
+      status: ApprovalStatus.approved
+    });
+    const { errors } = await performMutation(apolloClient, {
+      uuid: offers[ApplicantType.both].uuid
+    });
+    expect(errors).toEqualGraphQLErrorType(UnauthorizedError.name);
+  });
+
+  it("returns an error if the current user is a rejected applicant", async () => {
+    const { apolloClient } = await TestClientGenerator.applicant({
+      status: ApprovalStatus.rejected
+    });
+    const { errors } = await performMutation(apolloClient, {
+      uuid: offers[ApplicantType.both].uuid
+    });
+    expect(errors).toEqualGraphQLErrorType(UnauthorizedError.name);
+  });
+
+  it("returns an error if the current user is from a company pending", async () => {
+    const { apolloClient } = await TestClientGenerator.company();
+    const { errors } = await performMutation(apolloClient, {
+      uuid: offers[ApplicantType.both].uuid
+    });
+    expect(errors).toEqualGraphQLErrorType(UnauthorizedError.name);
+  });
+
+  it("returns an error if the current user is from a rejected company", async () => {
+    const { apolloClient } = await TestClientGenerator.company({ status: ApprovalStatus.rejected });
+    const { errors } = await performMutation(apolloClient, {
+      uuid: offers[ApplicantType.both].uuid
+    });
+    expect(errors).toEqualGraphQLErrorType(UnauthorizedError.name);
+  });
+
+  it("returns an error if the current user is from a rejected company", async () => {
+    const { apolloClient } = await TestClientGenerator.company({ status: ApprovalStatus.rejected });
+    const { errors } = await performMutation(apolloClient, {
+      uuid: offers[ApplicantType.both].uuid
+    });
+    expect(errors).toEqualGraphQLErrorType(UnauthorizedError.name);
+  });
```

Comment:
Este test esta repetido, aca seria:

`returns an error if the current user is from an aporoved company`

que es el que falta

Reply:
claro esta repetido pero no deberia tirar error si es from an approved company

Reply:
ah tenes razon, seria borrarlo entonces